src/main/java/net/dries007/tfc/objects/entity/animal/EntityGoatTFC.java

```diff
@@ -70,6 +75,61 @@ public void birthChildren()
         }
     }
 
+    @Override
+    public boolean isFood(@Nonnull ItemStack stack)
+    {
+        // Check for rotten
+        IFood cap = stack.getCapability(CapabilityFood.CAPABILITY, null);
+        if (!ConfigTFC.Animals.GOAT.acceptRotten && cap != null && cap.isRotten())
+        {
+            return false;
+        }
+        // Check if item is accepted
+        for (String input : ConfigTFC.Animals.GOAT.food)
```

Comment:
Firstly, this is *horrendously* gross. It's also duplicated verbatim all over various animals, which just makes it worse. It's using what looks to be a custom implementation of an `Ingredient`, and you're also doing the parsing of the actual ingredient on demand, every time.

An alternate proposal:
 - Change the config to a single string.
 - The format of this string must be a valid `Ingredient` (vanilla) json. (although obviously newlines removed because it's a config and all of these config values are hacked onto what really should be (and will be) json eventually).
 - Then, to parse this, use `gson` and call `CraftingHelper#getIngredient(JsonElement, JsonContext)` to return a vanilla `Ingredient` instance. Lazily cache this as part of the animal data, and invalidate it when configs change (there's an event for that).

Reply:
Copy from discord:

> I concede the first implementation i did for animal feeding list is gross, but putting a json in a config? Don't seem right to me, if we want to use json (and i prefer to do that in 1.15 for certain, although i would use the datapacks if possible), I prefer to use an actual json file, like i did for damage resistances
so, the counter-proposal is that, intead of a config, a json file containing all current animals food with a loader=animalFoodList

Thoughts?

Reply:
Done!